pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -865,7 +865,12 @@ def list(self, request, *args, **kwargs):
         """
         Returns paginated CollectionVersions list.
         """
-        queryset = self.filter_queryset(self.get_queryset())
+        queryset = self.get_queryset()
+
+        # prevent OOMKILL ...
+        queryset = queryset.only("pk", "content_ptr_id")
+
+        queryset = self.filter_queryset(queryset)
```

Comment:
It is unclear to me why `filter_queryset` may already OOM, because it should really only apply filters to the queryset, not evaluate it. And I don't see anything obvious in either inherited `filter_queryset` nor the `CollectionVersionFilter` class. Does the paginated version OOM too? Then I would suspect this fix will just by us some time until these two fields are able to fill the memory again. Approving anyway.

Reply:
Maybe add some of these thought to the comment to spark further investigation.